block_engine/src/block_engine.rs

```diff
@@ -342,13 +348,16 @@ impl BlockEngineRelayerHandler {
         exit: &Arc<AtomicBool>,
         aoi_cache_ttl_s: u64,
         address_lookup_table_cache: &DashMap<Pubkey, AddressLookupTableAccount>,
+        is_connected_to_block_engine: &Arc<AtomicBool>,
     ) -> BlockEngineResult<()> {
         let mut aoi_stream = subscribe_aoi_stream.into_inner();
         let mut poi_stream = subscribe_poi_stream.into_inner();
 
         // drain old buffered packets before streaming packets to the block engine
         while block_engine_receiver.try_recv().is_ok() {}
 
+        is_connected_to_block_engine.store(true, Ordering::Relaxed);
```

Comment:
should the this be based on HealthManager? eg https://github.com/jito-foundation/jito-relayer/blob/8bbb22c5b113cc71d43d39d073b1b2213d664771/relayer/src/health_manager.rs#L18

Reply:
health manager monitors rpc health (should prob be renamed such). you can still be connected to block engine and have unhealthy rpcs

Reply:
this is only for block engine, not for rpc health. the status is combined in the health webserver to come up with the final health state + other info

Reply:
sorry realized my comment is wrong, updated now